src/crtoolLocalStorage.js

```diff
@@ -0,0 +1,260 @@
+import C from "./js/business.logic/constants";
+
+const CHECK_FREQUENCY = 60e3;
+
+function redirectHome() {
+    window.location.href = C.START_PAGE_RELATIVE_URL;
+}
+
+/**
+ * A proxy for the localStorage API which periodically saves to the DB and
+ * can synchronize itself when the page loads.
+ */
+const ls = {
+    review: {},
+    dirty: false,
+
+    async init() {
+        const token = ls.getUrlParameter('token');
+        let review_id = token || localStorage.getItem('curriculumReviewId') || '';
+        if (!review_id) {
+            return redirectHome();
+        }
+
+        // If review_id found but a token is not set, update the URL
+        if (!token && review_id) {
+            const new_url = ls.updateUrlParameter(window.location.href, 'token', review_id);
+            window.history.pushState({ path: new_url }, '', new_url);
+        }
+
+        let db_review;
+        try {
+            db_review = await ls.fetchReviewFromServer(review_id);
+        } catch (e) {
+            console.error(e);
+            // Do not allow end users to set their own ID via the URL or localStorage.
+            localStorage.removeItem('crtool.' + review_id);
+            localStorage.removeItem('curriculumReviewId');
+            redirectHome();
+            return;
+        }
+
+        const ls_review = ls.loadReviewFromLocalStorage(review_id);
+
+        if (ls_review && ls_review.ls_modified_time > db_review.ls_modified_time) {
+            // Trust that newer local time means more up-to-date data
+            ls.review = ls_review;
+            ls.dirty = true;
+            localStorage.setItem('curriculumReviewId', review_id)
+        } else {
+            ls.review = db_review;
+            ls.dirty = false;
+            // Just bring local storage up to speed
+            localStorage.setItem('crtool.' + db_review.id, JSON.stringify(db_review));
+            localStorage.setItem('curriculumReviewId', db_review.id);
+        }
+
+        setTimeout(ls.saveIfDirty, ls.dirty ? 0 : CHECK_FREQUENCY);
+    },
+
+    isReady() {
+        return Boolean(ls.review.id);
+    },
+
+    saveReviewToLocalStorage() {
+        ls.dirty = true;
+        ls.review.ls_modified_time = (new Date()).toUTCString();
+        localStorage.setItem('crtool.' + ls.review.id, JSON.stringify(ls.review));
+    },
+
+    isValidReview(review) {
+        if (typeof review !== 'object' || review === null) {
+            return false;
+        }
+
+        for (const prop of ['id', 'last_updated', 'ls_modified_time']) {
```

Comment:
@mrclay reviews are initially created in python. ls_modified_time is created in JavaScript. So this property will not exist the first time a review is created and therefore will always fail the isValidReview() test.